lib/ansible/modules/files/ini_file.py

```diff
@@ -239,11 +241,20 @@ def do_ini(module, filename, section=None, option=None, value=None,
     if changed and not module.check_mode:
         if backup:
             backup_file = module.backup_local(filename)
-        ini_file = open(filename, 'w')
+
         try:
-            ini_file.writelines(ini_lines)
-        finally:
-            ini_file.close()
+            tmpfd, tmpfile = tempfile.mkstemp()
```

Comment:
In 2.5.0 we've added a way for modules to put temporary files into a directory that the controller would clean up in case where the module fails unexpectedly.  To make use of that here, use this:

``` python
    tmpfd, tmpfile = tempfile.mkstemp(dir=module.tmpdir)
```